src/componentRenderer.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { renderToString } from 'react-dom/server'
+import { uniqueId } from 'lodash'
+
+import {
+  ServerStyleSheet,
+  __DO_NOT_USE_OR_YOU_WILL_BE_HAUNTED_BY_SPOOKY_GHOSTS
+} from 'styled-components'
+
+// For Jest. See: https://github.com/styled-components/styled-components/issues/1692
+const { StyleSheet } = __DO_NOT_USE_OR_YOU_WILL_BE_HAUNTED_BY_SPOOKY_GHOSTS
+StyleSheet.reset(true)
+
+const modes = {
+  CLIENT: 'client',
+  SERVER: 'server'
+}
+
+export function componentRenderer(config) {
+  const {
+    mode = modes.SERVER,
+    modules,
+    serialize = x => x
+    //
+  } = config
+
+  const components = Object.keys(modules).reduce((moduleMap, moduleName) => {
+    const Component = modules[moduleName]
+
+    return {
+      ...moduleMap,
+
+      [moduleName]: props => {
+        if (mode === modes.SERVER) {
+          const mountId = uniqueId('stitch-component-')
+          const sheet = new ServerStyleSheet()
+          const html = renderToString(
+            sheet.collectStyles(<Component {...props} />)
+          )
+          const css = sheet.getStyleTags()
+          const markup = `
+            <div id=${mountId}>
+              ${css}
```

Comment:
Ok, I _think_ this answers my other question over in the spec for this (I'm an SSR noob when it comes to React/SC). So does this result in inline `<style>` tags right inside of the `<div>`?

Reply:
Yup, right inline. I think in a future version we could use the streaming api to optimize this slightly but for now this should be fine